apps/game_backend/lib/game_backend/matches.ex

```diff
@@ -2,21 +2,48 @@ defmodule GameBackend.Matches do
   @moduledoc """
   Matches
   """
+  alias GameBackend.Users
+  alias GameBackend.Users.Currencies
   alias Ecto.Multi
   alias GameBackend.Repo
   alias GameBackend.Matches.ArenaMatchResult
 
-  def create_arena_match_results(match_id, results) do
-    transaction = Multi.new()
+  def create_arena_match_results(results) do
+    {:ok, conrrency_config_json} =
+      Application.app_dir(:game_backend, "priv/currencies_rules.json")
+      |> File.read()
 
-    transaction =
-      Enum.reduce(results, transaction, fn result, transaction_acc ->
-        attrs = Map.put(result, "match_id", match_id)
-        changeset = ArenaMatchResult.changeset(%ArenaMatchResult{}, attrs)
-        transaction_acc = Multi.insert(transaction_acc, {:insert, attrs["user_id"]}, changeset)
-        transaction_acc
-      end)
+    currency_config =
+      Jason.decode!(conrrency_config_json)
 
-    Repo.transaction(transaction)
+    Enum.reduce(results, Multi.new(), fn result, transaction_acc ->
+      changeset = ArenaMatchResult.changeset(%ArenaMatchResult{}, result)
+      {:ok, google_user} = Users.get_google_user(result["user_id"])
+
+      amount_of_trophies = Currencies.get_amount_of_currency_by_name(google_user.user.id, "Trophies")
+
+      amount =
+        get_amount_of_trophies_to_modify(amount_of_trophies, result["position"], currency_config)
+
+      Multi.insert(transaction_acc, {:insert, result["user_id"]}, changeset)
+      |> Multi.run(
+        {:add_trophies_to, result["user_id"]},
+        fn _, _ ->
+          Currencies.add_currency_by_name!(google_user.user.id, "Trophies", amount)
+        end
+      )
+    end)
+    |> Repo.transaction()
+  end
+
+  def get_amount_of_trophies_to_modify(current_trophies, position, currencies_config) do
+    Enum.sort_by(currencies_config["rank"], fn %{"maximum_rank" => maximum} -> maximum end, :asc)
+    |> Enum.find(fn %{"maximum_rank" => maximum} ->
+      maximum > current_trophies
+    end)
+    |> case do
+      nil -> 0
```

Comment:
`nil` can't be a valid result, like this it will happen that if you have more than 9999 trophies you can no longer loose or win as it will always return 0 as the amount here

Reply:
right, a fix could be having an "infinite" rank and default to that instead of 0